compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1387,7 +1387,9 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
 
         self.note_obligation_cause(&mut err, &obligation);
         self.point_at_returns_when_relevant(&mut err, &obligation);
-
+        if obligation.predicate.references_error() {
+            err.delay_as_bug();
+        }
```

Comment:
Is this necessary for your fix?

Reply:
sorry. I forgot to remove these code after testing the function Esteban suggested.
I'm going to remove them soon.

Reply:
removed